Compilation error with clang on Linux: non-constant-expression cannot be narrowed

I've tested with libstdc++ and libc++, in Debug & Release build type, clang 3.9 to 11, same error:

```
-- The C compiler identification is Clang 11.1.0
-- The CXX compiler identification is Clang 11.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Conan: Compiler Clang>=8, checking major version 11
-- Conan: Checking correct version: 11
-- Conan: C++ stdlib: libc++
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DATAROOTDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR


-- Build files have been written to: /home/conan/.conan/data/laz-perf/2.0.0/bincrafters/testing/build/0e71bdb5ca512e4d54ff592784bdaead9275958d
Scanning dependencies of target lazperf
Scanning dependencies of target lazperf_s
[  2%] Building CXX object source_subfolder/cpp/lazperf/CMakeFiles/lazperf_s.dir/charbuf.cpp.o
[  5%] Building CXX object source_subfolder/cpp/lazperf/CMakeFiles/lazperf.dir/charbuf.cpp.o
[  8%] Building CXX object source_subfolder/cpp/lazperf/CMakeFiles/lazperf.dir/lazperf.cpp.o
[ 11%] Building CXX object source_subfolder/cpp/lazperf/CMakeFiles/lazperf_s.dir/lazperf.cpp.o
[ 14%] Building CXX object source_subfolder/cpp/lazperf/CMakeFiles/lazperf.dir/filestream.cpp.o
[ 17%] Building CXX object source_subfolder/cpp/lazperf/CMakeFiles/lazperf_s.dir/filestream.cpp.o
[ 20%] Building CXX object source_subfolder/cpp/lazperf/CMakeFiles/lazperf.dir/io.cpp.o
[ 23%] Building CXX object source_subfolder/cpp/lazperf/CMakeFiles/lazperf_s.dir/io.cpp.o
[ 26%] Building CXX object source_subfolder/cpp/lazperf/CMakeFiles/lazperf.dir/vlr.cpp.o
/home/conan/.conan/data/laz-perf/2.0.0/bincrafters/testing/build/0e71bdb5ca512e4d54ff592784bdaead9275958d/source_subfolder/cpp/lazperf/vlr.cpp:188:28: error: non-constant-expression cannot be narrowed from type '__uint32_t' (aka 'unsigned int') to 'uint8_t' (aka 'unsigned char') in initializer list [-Wc++11-narrowing]
    reserved{}, data_type{ htole32(1) }, options{}, name{}, unused{},
                           ^~~~~~~~~~
/usr/include/endian.h:70:22: note: expanded from macro 'htole32'
#  define htole32(x) __uint32_identity (x)
                     ^~~~~~~~~~~~~~~~~~~~~
/home/conan/.conan/data/laz-perf/2.0.0/bincrafters/testing/build/0e71bdb5ca512e4d54ff592784bdaead9275958d/source_subfolder/cpp/lazperf/vlr.cpp:188:28: note: insert an explicit cast to silence this issue
    reserved{}, data_type{ htole32(1) }, options{}, name{}, unused{},
                           ^~~~~~~~~~
                           static_cast<uint8_t>( )
/usr/include/endian.h:70:22: note: expanded from macro 'htole32'
#  define htole32(x) __uint32_identity (x)
                     ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
source_subfolder/cpp/lazperf/CMakeFiles/lazperf.dir/build.make:133: recipe for target 'source_subfolder/cpp/lazperf/CMakeFiles/lazperf.dir/vlr.cpp.o' failed
make[2]: *** [source_subfolder/cpp/lazperf/CMakeFiles/lazperf.dir/vlr.cpp.o] Error 1
make[1]: *** [source_subfolder/cpp/lazperf/CMakeFiles/lazperf.dir/all] Error 2
CMakeFiles/Makefile2:242: recipe for target 'source_subfolder/cpp/lazperf/CMakeFiles/lazperf.dir/all' failed
make[1]: *** Waiting for unfinished jobs....
[ 29%] Building CXX object source_subfolder/cpp/lazperf/CMakeFiles/lazperf_s.dir/vlr.cpp.o
[ 32%] Building CXX object source_subfolder/cpp/lazperf/CMakeFiles/lazperf_s.dir/detail/field_point10.cpp.o
/home/conan/.conan/data/laz-perf/2.0.0/bincrafters/testing/build/0e71bdb5ca512e4d54ff592784bdaead9275958d/source_subfolder/cpp/lazperf/vlr.cpp:188:28: error: non-constant-expression cannot be narrowed from type '__uint32_t' (aka 'unsigned int') to 'uint8_t' (aka 'unsigned char') in initializer list [-Wc++11-narrowing]
    reserved{}, data_type{ htole32(1) }, options{}, name{}, unused{},
                           ^~~~~~~~~~
/usr/include/endian.h:70:22: note: expanded from macro 'htole32'
#  define htole32(x) __uint32_identity (x)
                     ^~~~~~~~~~~~~~~~~~~~~
/home/conan/.conan/data/laz-perf/2.0.0/bincrafters/testing/build/0e71bdb5ca512e4d54ff592784bdaead9275958d/source_subfolder/cpp/lazperf/vlr.cpp:188:28: note: insert an explicit cast to silence this issue
    reserved{}, data_type{ htole32(1) }, options{}, name{}, unused{},
                           ^~~~~~~~~~
                           static_cast<uint8_t>( )
/usr/include/endian.h:70:22: note: expanded from macro 'htole32'
#  define htole32(x) __uint32_identity (x)
                     ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
source_subfolder/cpp/lazperf/CMakeFiles/lazperf_s.dir/build.make:133: recipe for target 'source_subfolder/cpp/lazperf/CMakeFiles/lazperf_s.dir/vlr.cpp.o' failed
make[2]: *** [source_subfolder/cpp/lazperf/CMakeFiles/lazperf_s.dir/vlr.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [source_subfolder/cpp/lazperf/CMakeFiles/lazperf_s.dir/all] Error 2
CMakeFiles/Makefile2:215: recipe for target 'source_subfolder/cpp/lazperf/CMakeFiles/lazperf_s.dir/all' failed
make: *** [all] Error 2
Makefile:148: recipe for target 'all' failed
```

--- a/cpp/lazperf/vlr.cpp
+++ b/cpp/lazperf/vlr.cpp
@@ -30,7 +30,6 @@
 
 #include <string>
 
-#include "portable_endian.hpp"
 #include "utils.hpp"
 #include "vlr.hpp"
 
@@ -185,7 +184,7 @@ std::vector<char> laz_vlr::data() const
 // EB VLR
 
 eb_vlr::ebfield::ebfield() :
-    reserved{}, data_type{ htole32(1) }, options{}, name{}, unused{},
+    reserved{}, data_type{1}, options{}, name{}, unused{},
     no_data{}, minval{}, maxval{}, scale{}, offset{}, description{}
 {}
 

